cmd/minishift/cmd/start_preflight_notwin.go

```diff
@@ -0,0 +1,53 @@
+// +build !windows
+
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+
+	"github.com/minishift/minishift/pkg/util/filehelper"
+)
+
+// check in the path for linux/darwin
+func checkVBoxInstalled() bool {
+	vboxCmd := "VBoxManage"
+	path, err := exec.LookPath(vboxCmd)
+	if err != nil {
+		return false
+	}
+	fi, _ := os.Stat(path)
+	if fi.Mode()&os.ModeSymlink != 0 {
+		path, err = os.Readlink(path)
+		if err != nil {
+			return false
+		}
+	}
+	// check in the default applications path in macOS
+	if runtime.GOOS == "darwin" {
```

Comment:
Where is this coming from? I'd argue this is not correct. And why does it matter? Is the central point not to find VBoxManage on the path?

Reply:
this check is to ensure that, if virtualbox in mac is removed by removing the virtualbox app from finder, the script `VBoxManage` is still on the path, but cannot access the original `VBoxManage` binary,  which is in the location `/Applications/VirtualBox.app/Contents/MacOS/VBoxManage` to care for that situation this check is added.

This prevents where we find the `VBoxManage` in path but cannot access the actual binary.
The right way to remove virtualbox is by using the unistaller in the dmg image.

Reply:
This implies it is in the default location. This does not have to be the case. I personally structure my /Application directory instead of flat out adding everything in there. This effectively would brake things on my machine. 

Given that install locations are often configurable and up to the user, I am against doing checks against fixed paths. Checks should be against required binaries on the PATH.

Reply:
If you want to make sure that _VBoxManage_ works, you can either check whether it is a symbolic link and its target exist or you can execute a known command, eg `VBoxManage -v`. Verify that there is a valid output. This is generic and works for Linux nicely as well, where the _VBosManage_ binary could be a symbolic link as well.

Reply:
in the machine i tried `VBoxManage' on the path was not a symbolic link but a shell script which would `exec` the `VBoxManage` binary inside Application.
It already checks for it on the path,  and checks if it is a symlink and follows it. 

I added this as another check to really make sure it is there, the situation you mentioned where it is installed in a different location will make this fail. Probably running a command like `VBoxManage -v` is the way to go.

Reply:
When `VBoxManage` is still on the path, this might have been caused by an issue in which the application got removed from `/Applications` on macOS, but not uninstalled. The included driver from Libmachine is actually wrong and uses this as a fallback case.

Reply:
Let's execute 'VBoxManage -v' then. That's the interaction point between the driver and VirtualBox. Making assumptions about the install location is imo wrong.

Reply:
So what is the user chooses user only install (not for all users). In this case ~/Applications is used.

Reply:
> I believe that when xhyve is selected, the driver will eventually fail as it finds VBoxManage... right? This is the behaviour I saw

Fair enough. I am ok with trying to verify whether the VirtualBox install is working. However, this should be done in a way which is not too restrictive and makes wrong assumptions.